### PR TITLE
Feature/add lishid suggestions

### PR DESCRIPTION
**Pull request by trey-wallis**
- Replaced read with cached read for improved performance
- Added missing await
- Replaced `leaf.duplicate()` + `leaf.detach()`with MarkdownPreviewView.rerender
- Updated hashing with id based system

### CODE REVIEW

**Comment by trey-wallis** [comment]
No longer needed
